### PR TITLE
Automation: Monitor tests

### PR DESCRIPTION
**Pull request by psiinon**
Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

**Comment by psiinon** [comment]
Still todo - implement for active scan and change the GUI to only allow monitor tests to be added to jobs that support them.

**Comment by psiinon** [comment]
Seems to work locally in Eclipse but have reproduces a failure on the command line...

**Comment by psiinon** [comment]
Tests passed! 🎉

**Comment by psiinon** [comment]
Updated the spider and spiderAjax add-on jobs too.

File: addOns/automation/src/main/java/org/zaproxy/addon/automation/AutomationJob.java [hunk 1]
```diff
@@ -684,4 +729,12 @@ public User getUser(String username, AutomationProgress progress) {
         }
         return user;
     }
+
+    public void sleep(long millis) {
```
Comment (thc202) [inline addOns/automation/src/main/java/org/zaproxy/addon/automation/AutomationJob.java:733]: `protected`

File: addOns/automation/src/main/java/org/zaproxy/addon/automation/gui/MonitorTestDialog.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+package org.zaproxy.addon.automation.gui;
+
+import java.util.Arrays;
+import org.parosproxy.paros.view.View;
+import org.zaproxy.addon.automation.tests.AbstractAutomationTest.OnFail;
+import org.zaproxy.addon.automation.tests.AutomationMonitorTest;
+import org.zaproxy.zap.utils.DisplayUtils;
+import org.zaproxy.zap.view.StandardFieldsDialog;
+
+public class MonitorTestDialog extends StandardFieldsDialog {
```
Comment (thc202) [inline addOns/automation/src/main/java/org/zaproxy/addon/automation/gui/MonitorTestDialog.java:29]: Needs `@SuppressWarnings("serial")`.

File: addOns/automation/src/main/java/org/zaproxy/addon/automation/gui/StatisticTestDialog.java [hunk 3]
```diff
@@ -27,7 +27,6 @@
 import org.zaproxy.zap.utils.DisplayUtils;
 import org.zaproxy.zap.view.StandardFieldsDialog;
 
-@SuppressWarnings("serial")
```
Comment (thc202) [inline addOns/automation/src/main/java/org/zaproxy/addon/automation/gui/StatisticTestDialog.java:30]: Revert.

**Comment by psiinon** [comment]
Thanks - think I've fixed all of those..

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

File: addOns/automation/src/main/java/org/zaproxy/addon/automation/gui/MonitorTestDialog.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2021 The ZAP Development Team
```
Comment (kingthorin) [inline addOns/automation/src/main/java/org/zaproxy/addon/automation/gui/MonitorTestDialog.java:6]: 2022

Comment (psiinon) [reply addOns/automation/src/main/java/org/zaproxy/addon/automation/gui/MonitorTestDialog.java]: Good catch

**Review by kingthorin — APPROVED** [review]

**Pull request merged**